internal/contourconfig/contourconfiguration.go

```diff
@@ -17,10 +17,147 @@ import (
 	"fmt"
 	"time"
 
+	"github.com/imdario/mergo"
 	contour_api_v1alpha1 "github.com/projectcontour/contour/apis/projectcontour/v1alpha1"
 	"github.com/projectcontour/contour/internal/timeout"
+	"k8s.io/utils/pointer"
 )
 
+func UIntPtr(val uint) *uint {
+	return &val
+}
+
+func UInt32Ptr(val uint32) *uint32 {
+	return &val
+}
+
+func OverlayOnDefaults(spec contour_api_v1alpha1.ContourConfigurationSpec) (contour_api_v1alpha1.ContourConfigurationSpec, error) {
+	res := Defaults()
+
+	if err := mergo.Merge(&res, spec, mergo.WithOverride); err != nil {
+		return contour_api_v1alpha1.ContourConfigurationSpec{}, err
+	}
+
+	return res, nil
+}
+
+func Defaults() contour_api_v1alpha1.ContourConfigurationSpec {
+	return contour_api_v1alpha1.ContourConfigurationSpec{
+		XDSServer: &contour_api_v1alpha1.XDSServerConfig{
+			Type:    contour_api_v1alpha1.ContourServerType,
+			Address: "0.0.0.0",
+			Port:    8001,
+			TLS: &contour_api_v1alpha1.TLS{
+				CAFile:   "/certs/ca.crt",
+				CertFile: "/certs/tls.crt",
+				KeyFile:  "/certs/tls.key",
+				Insecure: pointer.Bool(false),
+			},
+		},
+		Ingress: &contour_api_v1alpha1.IngressConfig{
+			ClassNames:    nil,
+			StatusAddress: "",
+		},
+		Debug: &contour_api_v1alpha1.DebugConfig{
+			// TODO verify the address/port defaults, this is a change from current
+			// but matches the pkg/config defaults.
+			Address:                 "127.0.0.1",
+			Port:                    6060,
+			DebugLogLevel:           contour_api_v1alpha1.InfoLog,
+			KubernetesDebugLogLevel: UIntPtr(0),
+		},
+		Health: &contour_api_v1alpha1.HealthConfig{
+			Address: "0.0.0.0",
+			Port:    8000,
+		},
+		Envoy: &contour_api_v1alpha1.EnvoyConfig{
+			Listener: &contour_api_v1alpha1.EnvoyListenerConfig{
+				UseProxyProto:             pointer.Bool(false),
+				DisableAllowChunkedLength: pointer.Bool(false),
+				DisableMergeSlashes:       pointer.Bool(false),
+				ConnectionBalancer:        "",
+				TLS: &contour_api_v1alpha1.EnvoyTLS{
+					MinimumProtocolVersion: "1.2",
+					CipherSuites: []contour_api_v1alpha1.TLSCipherType{
```

Comment:
nit: could we consolidate with https://github.com/projectcontour/contour/blob/6add750633e3b8d20890d10a92e65e033df45b16/pkg/config/ciphersuites.go#L33 maybe?